src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -1183,27 +1206,40 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 	}
 
 	// put thinking parts inside a pinned part. commented out for now.
-	// private shouldPinPart(part: IChatRendererContent, element?: IChatResponseViewModel): boolean {
-
-	// 	if ((part.kind === 'toolInvocation' || part.kind === 'toolInvocationSerialized') && element) {
-	// 		// Explicit set of tools that should be pinned when there has been thinking
-	// 		const specialToolIds = new Set<string>([
-	// 			'copilot_searchCodebase',
-	// 			'copilot_searchWorkspaceSymbols',
-	// 			'copilot_listCodeUsages',
-	// 			'copilot_think',
-	// 			'copilot_findFiles',
-	// 			'copilot_findTextInFiles',
-	// 			'copilot_readFile',
-	// 			'copilot_listDirectory',
-	// 			'copilot_getChangedFiles',
-	// 		]);
-	// 		const isSpecialTool = specialToolIds.has(part.toolId);
-	// 		return isSpecialTool && this._hasHitThinking;
-	// 	}
-
-	// 	return false;
-	// }
+	private shouldPinPart(part: IChatRendererContent, element?: IChatResponseViewModel): boolean {
+
+		if ((part.kind === 'toolInvocation' || part.kind === 'toolInvocationSerialized') && element) {
+			// Explicit set of tools that should be pinned when there has been thinking
+			const specialToolIds = new Set<string>([
+				'copilot_searchCodebase',
```

Comment:
If we keep this kind of approach, we would want another way to identify the tools to group. A tool tag might be one. Think about MCP/extension tools too.

Reply:
not for this PR, but I'm leaning more and more that if the tool doesn't require me to interact, pin it